cmd/cloud-controller-manager/nodeipamcontroller.go

```diff
@@ -126,17 +114,7 @@ func startNodeIpamController(initContext app.ControllerInitContext, ccmConfig *c
 		}
 	}
 
-	var nodeCIDRMaskSizeIPv4, nodeCIDRMaskSizeIPv6 int
-	if utilfeature.DefaultFeatureGate.Enabled(features.IPv6DualStack) {
-		// only --node-cidr-mask-size-ipv4 and --node-cidr-mask-size-ipv6 supported with dual stack clusters.
-		// --node-cidr-mask-size flag is incompatible with dual stack clusters.
-		nodeCIDRMaskSizeIPv4, nodeCIDRMaskSizeIPv6, err = setNodeCIDRMaskSizesDualStack(nodeIPAMConfig)
-	} else {
-		// only --node-cidr-mask-size supported with single stack clusters.
-		// --node-cidr-mask-size-ipv4 and --node-cidr-mask-size-ipv6 flags are incompatible with dual stack clusters.
-		nodeCIDRMaskSizeIPv4, nodeCIDRMaskSizeIPv6, err = setNodeCIDRMaskSizes(nodeIPAMConfig)
-	}
-
+	nodeCIDRMaskSizeIPv4, nodeCIDRMaskSizeIPv6, err := setNodeCIDRMaskSizesDualStack(nodeIPAMConfig, clusterCIDRs)
```

Comment:
ah, this code was broken before; it got fixed in kube-controller-manager but not here. You should just copy the kube-controller-manager fix here (https://github.com/kubernetes/kubernetes/pull/90439/files). You don't need to change `setNodeCIDRMaskSizesDualStack` or `setNodeCIDRMaskSizes`, all that needs to be done is to pick between them based on the actual configuration of the cluster rather than the feature gate.

Reply:
I actually think the code in `cmd/kube-controller-manager/app/core.go` needs to change and read correct flag similar to what we have done in `cmd/cloud-controller-manager/nodeipamcontroller.go` notice how if the cluster is IPv6 we read a different flag.

thoughts?

Reply:
I modified the logic. In a separate commit. PTAL.